### PR TITLE
PR for unobtrusive notification feature

### PR DESCRIPTION
**Pull request by share-with-me**
This PR displays an unobtrusive notification in the event that the `requests` take more than a threshold time. This demonstrates the feature but is blocked on the PR - POST v/s GET for requests. Simply running the code as is will demonstrate how the notification may look. I make use of `sessionStorage` to store the number of calls and the `cumulativeTime` for the calls.Thus, it does not show any notification when the cookies are disabled. A lot of code change is common across the two PRs and thus I was waiting to merge that PR before creating this one. However, I am creating this PR now.

### CODE REVIEW

File: assets/css/style.css [hunk 1]
```diff
@@ -74,6 +74,22 @@ html[dir=rtl] .clearButton {
...
+    height: 100px;
+    font-size: 12px;
+    z-index: 9999999;
+    top: 20px;
+    right: 20px;
+    position: fixed;
+    border: 8px solid #FFF;
+}
+
+.info-message .fa-times:hover{
```
Comment (sushain97) [inline assets/css/style.css]: Missing spacing.

File: assets/css/style.css [hunk 2]
```diff
@@ -74,6 +74,22 @@ html[dir=rtl] .clearButton {
     left: 18px;
 }
 
+.info-message {
```
Comment (sushain97) [inline assets/css/style.css]: Use camel case. If there's only a single one, use an ID.

File: index.html.in [hunk 3]
```diff
@@ -531,5 +531,9 @@
             <div class="lead text-warning well well-sm" data-text="Enable_JS_Warning"></div>
             <noscript><p><img src="@include_piwik_url@/piwik.php?idsite=@include_piwik_siteid@" style="border: 0;" alt=""></p></noscript>
         </div>
+        <div class="info-message well" style="margin: 8px; padding: 8px;">
```
Comment (sushain97) [inline index.html.in]: Does this need to be hidden on mobile? Don't use inline styles.

File: index.html.in [hunk 4]
```diff
@@ -531,5 +531,9 @@
...
+          <h4 style="margin: 0px; padding: 0px;">Inconvenience<i class="fa fa-times pull-right"></i></h4>
+          <p>We are highly sorry for the inconvenience caused but our server appears loaded now. Please try again later. <br/> In order to set apertium locally, please refer to <a href = "http://wiki.apertium.org/wiki/Apertium-apy" target="_blank"> this </a> link.
```
Comment (sushain97) [inline index.html.in]: Use the localisation framework.

File: index.html.in (hunk 3 above)
Comment (share-with-me) [inline index.html.in]: Um, should we hide it in mobile? Would ut take too much of screen space? ( leading to bad UX? )

File: assets/css/style.css (hunk 1 above)
Comment (share-with-me) [inline assets/css/style.css]: Noted

File: index.html.in (hunk 4 above)
Comment (share-with-me) [inline index.html.in]: Would start by localising to `eng` and `spa`!

**Comment by sushain97** [comment]
Don't forget to rebase this one! The conflicts abound right now :(

**Comment by share-with-me** [comment]
@sushain97 , I know :( . Actually, what I did is, I knew there could be conflicts so I manually copied the file contents of the files that you had modified as per the commit named `style fixes`. The `diff` it shows currently is between the `goavki/master` and `share-with-me/gsoc-unobtrusive-notification` right? But you already merged a lot of code into the `master`. So why a lot of this code appears in `diff`?

File: assets/strings/spa.json [hunk 5]
```diff
@@ -64,5 +64,7 @@
     "Contact": "Contactar",
     "Documentation": "Documentación",
     "About_Apertium": "Acerca de Apertium",
-    "Apertium_Downloads": "Descargas de Apertium"
+    "Apertium_Downloads": "Descargas de Apertium",
+    "Unobtrusive_Title": "Inconveniencia",
+    "Unobtrusive_Warning": "Somos altamente tristes para la inconveniencia causada pero nuestro servidor aparece cargado ahora. Para poner apertium localmente, complacer referir a esto <a href='http://wiki.apertium.org/wiki/Apertium-apy' target='_blank'> enlace</a>."
```
Comment (sushain97) [inline assets/strings/spa.json]: Do you speak Spanish at a ~native level?

**Comment by sushain97** [comment]
> So why a lot of this code appears in diff?

I don't know. I would either merge or rebase.

File: assets/strings/spa.json (hunk 5 above)
Comment (share-with-me) [inline assets/strings/spa.json]: > Do you speak Spanish at a ~native level?

Nope, this is for demo purposes only! I would remove this. Would keep only english.

**Comment by share-with-me** [comment]
@sushain97 , Um, I don't understand. I rebased with `master` and then applied the changes. For instance, why does the code `#backToTop` or `callApy()` appear in the `git diff` when it is already merged in the master? When I run `git diff` locally as `git diff master..gsoc-unobtrusive-notification`, it shows only the code snippet that belongs to `unobtrusive` feature.

**Comment by sushain97** [comment]
I'm not sure what Git is doing. Feel free to just create a new branch and copy your changes in, i.e. the simplest way out of this.

**Comment by share-with-me** [comment]
Thanks @sushain97 . This is exactly what I was gonna ask next. On it! Would it also make sense if I create a different PR through that branch altogether?! (and close this one?! gonna be very easy for me :P)

**Comment by sushain97** [comment]
> On it! Would it also make sense if I create a different PR through that branch altogether?! (and close this one?! gonna be very easy for me :P)

Sure. Up to you!

**Comment by share-with-me** [comment]
Closing this PR here on account of a lot rebase and merge conflicts. Continuing it in #139.

**Pull request closed without merging**